### PR TITLE
feat (batcher): dont retry + gas escalator + sequential txs

### LINKED ISSUE #677: feat (batcher): don't retry when not needed
Only retry submitting batch when error is nonce too low or not enough gas. Don't retry on business logic reverts because it's unlikely it will be fixed on retry.

### PR DESCRIPTION
**Pull request by taturosati**
- Add RetryClient to retry on http errors
- Use gas escalator to prevent txs stuck on mempool
- Add lock to write on blockchain to prevent nonce errors (batcher now waits for tx success / failure before sending next tx)

# To test

Normal flow

### CODE REVIEW

File: batcher/aligned-batcher/src/eth/mod.rs [hunk 1]
```diff
@@ -19,15 +15,31 @@ pub struct BatchVerified {
     pub batch_merkle_root: [u8; 32],
 }
 
-pub type BatcherPaymentService =
-    BatcherPaymentServiceContract<SignerMiddleware<Provider<Http>, Wallet<SigningKey>>>;
+pub type BatcherPaymentService<T> = BatcherPaymentServiceContract<
```
Comment (entropidelic) [inline batcher/aligned-batcher/src/eth/mod.rs]: do we need it to be generic over T? The batcher payment service only seems to use `Http` for the `Provider` generic

Comment (taturosati) [reply batcher/aligned-batcher/src/eth/mod.rs]: Changed on 5f8cf3ab09970f06aa30c644cef65333999c5f33

File: batcher/aligned-sdk/src/eth/batcher_payment_service.rs
Comment (entropidelic) [inline batcher/aligned-sdk/src/eth/batcher_payment_service.rs:1]: seems like the code in this file is duplicated

File: batcher/aligned-batcher/src/eth/mod.rs [hunk 2]
```diff
@@ -1,93 +1,52 @@
...
+    SignerMiddleware<GasEscalatorMiddleware<Provider<RetryClient<Http>>>, Wallet<SigningKey>>,
+>;
 
-pub fn get_provider(eth_rpc_url: String) -> Result<Provider<Http>, anyhow::Error> {
-    Provider::<Http>::try_from(eth_rpc_url).map_err(|err| anyhow::anyhow!(err))
-}
+const MAX_RETRIES: u32 = 15;
+const INITIAL_BACKOFF: u64 = 1000;
+const GAS_MULTIPLIER: f64 = 1.125;
+const GAS_ESCALATOR_INTERVAL: u64 = 12; // seconds
```
Comment (entropidelic) [inline batcher/aligned-batcher/src/eth/mod.rs]: let's add some documentation here about these constants

File: batcher/aligned-batcher/src/lib.rs [hunk 3]
```diff
@@ -566,19 +575,77 @@ impl Batcher {
...
-            payment_service,
-            *batch_merkle_root,
+        match self
+            .create_new_task(
+                *batch_merkle_root,
+                batch_data_pointer,
+                leaves,
+                signatures,
+                AGGREGATOR_COST.into(), // FIXME(uri): This value should be read from aligned_layer/contracts/script/deploy/config/devnet/batcher-payment-service.devnet.config.json
+                gas_per_proof.into(), //FIXME(uri): This value should be read from aligned_layer/contracts/script/deploy/config/devnet/batcher-payment-service.devnet.config.json
```
Comment (entropidelic) [inline batcher/aligned-batcher/src/lib.rs]: remove these FIXME comments, they do not apply anymore

File: batcher/aligned-sdk/src/eth/batcher_payment_service.rs
Comment (taturosati) [reply batcher/aligned-sdk/src/eth/batcher_payment_service.rs:1]: Code in aligned-batcher is a wrapper around this, this is without wallet just for read

File: batcher/aligned-batcher/src/lib.rs (hunk 3 above)
Comment (taturosati) [reply batcher/aligned-batcher/src/lib.rs]: removed on e45d40438dc3271f447adb9c402aa4b261e9f757

File: batcher/aligned-batcher/src/eth/mod.rs (hunk 2 above)
Comment (taturosati) [reply batcher/aligned-batcher/src/eth/mod.rs]: added on e45d40438dc3271f447adb9c402aa4b261e9f757

File: batcher/aligned-sdk/src/eth/batcher_payment_service.rs
Comment (entropidelic) [reply batcher/aligned-sdk/src/eth/batcher_payment_service.rs:1]: ah ok, makes sense. But do we need the generic here?

Comment (taturosati) [reply batcher/aligned-sdk/src/eth/batcher_payment_service.rs:1]: not needed you are right, changed on 91687b18ac2affe4838e45d5ba593f7582cc08ae

**Review by entropidelic — DISMISSED** [review]

**Review by MauroToscano — APPROVED** [review]

**Review by NicolasRampoldi — APPROVED** [review]

**Pull request merged**